src/middleware.js

```diff
@@ -65,3 +66,44 @@ function getRequestInfo(ctx) {
     size,
   };
 }
+
+const BLACKLIST = /token|password|secret|hash|jwt/i;
+
+function getRequestParams(ctx, options) {
+  if (ctx.status >= 400) {
+    const { body, query } = ctx.request;
+    return {
+      requestBody: strip(body, options),
+      requestQuery: strip(query, options),
+    };
+  } else {
+    return {};
+  }
+}
```

Comment:
nice, i would make it perhaps only 500?

Reply:
Part of the point of this is to be able to get insights into any request failures so 400 level is just as important, maybe more so if we're doing probably error handling, which we tend to